Sources/MessagingInApp/Gist/Managers/MessageQueueManager.swift

```diff
@@ -123,12 +152,22 @@ class MessageQueueManager {
             }
         }
 
-        if let elementId = message.gistProperties.elementId {
-            Logger.instance.info(message: "Embedding message with Element Id \(elementId)")
-            Gist.shared.embedMessage(message: message, elementId: elementId)
+        if message.isInlineMessage {
+            // If message has an element id, it's meant to be shown inline.
+            // Inline Views show inline messages by getting messages stored in the local queue on device.
+            // So, add the message to the local store and when inline Views are constructed, they will check the store.
+
+            addMessageToLocalStore(message: message)
+
+            // In a future PR, we will want to notify all currently visible inline Views that new messages are available in local store.
+            //
+            // At that time, we may decide we do not need these lines anymore. Keeping them in until we implement this notify piece.
+            //            Logger.instance.info(message: "Found a message meant to be shown inline. Element Id \(elementId)")
+            //            Gist.shared.embedMessage(message: message, elementId: elementId)
+
```

Comment:
Modified logic here to add all inline messages to local queue.

Reply:
// TODO: I need to verify that when page rule triggers and SDK checks if we need to display any modal messages it filters out inline messages.